runtime/fastly/host-api/host_api.cpp

```diff
@@ -202,6 +204,191 @@ Response make_response(fastly::fastly_host_http_response &resp) {
   return Response{HttpResp{resp.f0}, HttpBody{resp.f1}};
 }
 
+template <auto header_names_get>
+Result<std::vector<HostString>> generic_get_header_names(auto handle) {
+  Result<std::vector<HostString>> res;
+
+  fastly::fastly_world_list_string ret;
+  fastly::fastly_host_error err;
+  fastly::fastly_world_string *strs = static_cast<fastly::fastly_world_string *>(
+      cabi_malloc(LIST_ALLOC_SIZE * sizeof(fastly::fastly_world_string), 1));
```

Comment:
What do you think about pushing into the result vector in the main loop, instead of keeping the separate `strs` buffer that's free'd later? That would avoid a memory leak with `strs` when there's an error in the conditional on line 222.

Reply:
As discussed, let's handle this in a follow-up PR.